src/backend/common/ArrayInfo.hpp

```diff
@@ -79,10 +83,24 @@ class ArrayInfo {
                    This is then used in the unified backend to check mismatched arrays.");
     }
 
-    // Copy constructors are deprecated if there is a
-    // user-defined destructor in c++11
     ArrayInfo()                       = default;
     ArrayInfo(const ArrayInfo& other) = default;
+    ArrayInfo(ArrayInfo&& other)      = default;
+
+    ArrayInfo& operator=(ArrayInfo other) noexcept {
```

Comment:
Shouldn't this take `ArrayInfo &&other` if is meant to be move assignment operator ?

Reply:
This is an implementation of the "Rule of 4 and a half". It handles both copy and move assignment using this approach.

Reply:
I don't much about these rules by names. I will read up.